ui/src/app/core/settings.service.ts

```diff
@@ -0,0 +1,58 @@
+import {Injectable} from '@angular/core';
+import {AuthService} from '../core/auth.service';
+import {CapabilitiesService} from "./capabilities.service";
+import {DisplayField} from "../shared/model/DisplayField";
+
+type Settings = {
+  // Unset for basic auth users
+  user: string
+  // Unset for non-project-based job manager
+  project: string
```

Comment:
Should it scope by project first and then by user or vice versa?

Reply:
I would opt to only store settings for the most recent user, i.e. if we see a new user just reset to the default settings, rather than trying to store the settings for all users the have logged into a machine. 

If that's a use case we expect (maybe users often have different accounts they sign in with), then I would probably scope by user first though I guess it doesn't matter too much.

Reply:
In terms of scoping by user vs. project first, I think either one will work fine! Although project-first may make more sense for non-project-based backends, otherwise each user will have a project key that isn't really necessary